src/components/Chart.tsx

```diff
@@ -0,0 +1,156 @@
+/** Load dependencies */
+import React from "react";
+import { Chart as ChartJS } from "chart.js";
+import "chartjs-plugin-datalabels";
+
+/** Load types */
+import { Metric } from "../types";
+
+/** Types */
+type Context =
+  | string
+  | CanvasRenderingContext2D
+  | HTMLCanvasElement
+  | ArrayLike<CanvasRenderingContext2D | HTMLCanvasElement>;
+
+type RefContext =
+  | string
+  | ((instance: HTMLCanvasElement | null) => void)
+  | React.RefObject<HTMLCanvasElement>
+  | null
+  | undefined;
+
+export type ChartTypes =
+  | "line"
+  | "bar"
+  | "radar"
+  | "pie"
+  | "doughnut"
+  | "polarArea";
+
+/** Interfaces */
+interface Datalabels {
+  anchor: string;
+  align: string;
+  formatter: (x: number) => number;
+  font: {
+    weight: string;
+  };
+}
+
+interface ChartProps {
+  className?: string;
+  data: {
+    labels: Metric["label"][];
+    values: Metric["value"][];
+  };
+  types?: ChartTypes[];
+}
+
+/** Component */
+const Chart: React.FunctionComponent<ChartProps> = ({
+  className,
+  data,
+  types,
+}: ChartProps) => {
+  const [chartType, setChartType] = React.useState<ChartTypes>("bar");
+  const ctx: React.MutableRefObject<Context> = React.useRef<Context>("");
+
+  React.useEffect(() => {
+    const chart: Context = ctx.current;
+
+    const currentChart = new ChartJS(chart, {
+      plugins: {
+        datalabels: {
+          anchor: "end",
+          align: "top",
+          formatter: Math.round,
+          font: {
+            weight: "bold",
+          },
+        } as Datalabels,
+      } as any,
+      type: chartType,
+      data: {
+        labels: data.labels,
+        datasets: [
+          {
+            data: data.values,
+            backgroundColor: [
+              "rgba(255, 99, 132, 0.2)",
+              "rgba(54, 162, 235, 0.2)",
+              "rgba(255, 206, 86, 0.2)",
+              "rgba(75, 192, 192, 0.2)",
+              "rgba(153, 102, 255, 0.2)",
+              "rgba(255, 159, 64, 0.2)",
+              "rgba(54, 179, 100, 0.2)",
+            ],
+            borderColor: [
+              "rgba(255, 99, 132, 1)",
+              "rgba(54, 162, 235, 1)",
+              "rgba(255, 206, 86, 1)",
+              "rgba(75, 192, 192, 1)",
+              "rgba(153, 102, 255, 1)",
+              "rgba(255, 159, 64, 1)",
+              "rgba(54, 179, 100, 1)",
+            ],
```

Comment:
This is kind of magic :-)

- Why an array of 7 colors? I am wondering if there is a way to make it clear how the data relates to each column setup? 
- My IDE shows the  color behind the rgb values but could
 we do this better?